src/main/scala/cromwell/binding/values/WdlObject.scala

```diff
@@ -1,14 +1,84 @@
 package cromwell.binding.values
 
 import cromwell.binding.Call
-import cromwell.binding.types.{WdlCallOutputsObjectType, WdlObjectType}
+import cromwell.binding.types._
+import cromwell.util.FileUtil
+
+import scala.util.{Failure, Success, Try}
 
 trait WdlObjectLike {
   def value: Map[String, WdlValue]
 }
 
+object WdlObject {
+
+  def coerceObject(m: Map[String, String]): WdlObject = {
+    val coerced = WdlMap.coerceMap(m, WdlMapType(WdlStringType, WdlAnyType)).value map {
+      case (k, v) => k.valueString -> v
+    }
+
+    WdlObject(coerced)
+  }
+
+  def fromTsv(tsv: String): Try[Array[WdlObject]] = {
+    FileUtil.parseTsv(tsv) match {
+      case Success(table) if table.isEmpty => Failure(new UnsupportedOperationException("TSV file was empty or could not be parsed."))
+      case Success(table) if table.length < 2 => Failure(new UnsupportedOperationException("TSV must be 2 rows (or more) to convert to an Object (Array[Object])"))
+      case Success(table) => Try {
+        table.tail map { line => coerceObject((table.head zip line).toMap) }
+      }
+      case Failure(e) => Failure(e)
+    }
+  }
+
+  def tsvSerializeArray(input: Seq[WdlObject]) = {
```

Comment:
So this may not be a problem, and it may be outside the scope of this ticket, but generating entire files in memory is generally a slightly questionable idea... especially if we're talking about several GB/TB/PB of data...